Problem in connect to remote stan server

Hi, i am trying to connect go stan client to remote stan server and server tell me something like that 
" 2016/07/16 21:48:31 Can't connect: nats: no servers available for connection.
Make sure a NATS Streaming Server is running at: nats://10.0.1.60:4222
"

steps what i am do:
1. make file seed.conf who look like 
"
listen: 0.0.0.0:4222
http: 8222
 "

2 start  server with command:
"./nats-streaming-server -c seed.conf"
who tell me
"
[6763] 2016/07/16 22:52:05.256046 [INF] Starting nats-streaming-server[test-cluster] version 0.2.0
[6763] 2016/07/16 22:52:05.256521 [INF] Starting nats-server version 0.9.0.beta
[6763] 2016/07/16 22:52:05.256544 [INF] Starting http monitor on :8222
[6763] 2016/07/16 22:52:05.256643 [INF] Listening for client connections on 0.0.0.0:4222
"
3. I want to send message to server from other computer whith standart instruction on site
go run stan-pub.go -c "nats://10.0.1.60:4222" foo "msg one" 
and take a message : 
Make sure a NATS Streaming Server is running at: nats://10.0.1.60:4222

what am i doing wrong?

And nwo one problem,  i have a code like in archive
[main.go.zip](https://github.com/nats-io/go-nats-streaming/files/369332/main.go.zip)


but it tells me some thing like this

:!
!!!
panic: too many subscriptions per channel

I need more doc, what subscriptions  It has
 mean.
It working with only one thread and only one subscriber.
I am feeling myself like a stuped newbie (((

--- a/examples/stan-pub.go
+++ b/examples/stan-pub.go
@@ -31,14 +31,13 @@ func usage() {
 }
 
 func main() {
-	opts := stan.DefaultOptions
-
 	var clusterID string
 	var clientID string
 	var async bool
+	URL := stan.DefaultNatsURL
 
-	flag.StringVar(&opts.NatsURL, "s", stan.DefaultNatsURL, "The nats server URLs (separated by comma)")
-	flag.StringVar(&opts.NatsURL, "server", stan.DefaultNatsURL, "The nats server URLs (separated by comma)")
+	flag.StringVar(&URL, "s", stan.DefaultNatsURL, "The nats server URLs (separated by comma)")
+	flag.StringVar(&URL, "server", stan.DefaultNatsURL, "The nats server URLs (separated by comma)")
 	flag.StringVar(&clusterID, "c", "test-cluster", "The NATS Streaming cluster ID")
 	flag.StringVar(&clusterID, "cluster", "test-cluster", "The NATS Streaming cluster ID")
 	flag.StringVar(&clientID, "id", "stan-pub", "The NATS Streaming client ID to connect with")
@@ -56,9 +55,9 @@ func main() {
 		usage()
 	}
 
-	sc, err := stan.Connect(clusterID, clientID)
+	sc, err := stan.Connect(clusterID, clientID, stan.NatsURL(URL))
 	if err != nil {
-		log.Fatalf("Can't connect: %v.\nMake sure a NATS Streaming Server is running at: %s", err, opts.NatsURL)
+		log.Fatalf("Can't connect: %v.\nMake sure a NATS Streaming Server is running at: %s", err, URL)
 	}
 	defer sc.Close()
 

--- a/examples/stan-pub.go
+++ b/examples/stan-pub.go
@@ -34,7 +34,7 @@ func main() {
 	var clusterID string
 	var clientID string
 	var async bool
-	URL := stan.DefaultNatsURL
+	var URL string
 
 	flag.StringVar(&URL, "s", stan.DefaultNatsURL, "The nats server URLs (separated by comma)")
 	flag.StringVar(&URL, "server", stan.DefaultNatsURL, "The nats server URLs (separated by comma)")

--- a/examples/stan-sub.go
+++ b/examples/stan-sub.go
@@ -54,8 +54,7 @@ func main() {
 	var durable string
 	var qgroup string
 	var unsubscribe bool
-
-	URL := DefaultNatsURL
+	var URL string
 
 	//	defaultID := fmt.Sprintf("client.%s", nuid.Next())
 

--- a/examples/stan-sub.go
+++ b/examples/stan-sub.go
@@ -44,8 +44,6 @@ func printMsg(m *Msg, i int) {
 }
 
 func main() {
-	opts := DefaultOptions
-
 	var clusterID string
 	var clientID string
 	var showTime bool
@@ -57,10 +55,12 @@ func main() {
 	var qgroup string
 	var unsubscribe bool
 
+	URL := DefaultNatsURL
+
 	//	defaultID := fmt.Sprintf("client.%s", nuid.Next())
 
-	flag.StringVar(&opts.NatsURL, "s", DefaultNatsURL, "The nats server URLs (separated by comma)")
-	flag.StringVar(&opts.NatsURL, "server", DefaultNatsURL, "The nats server URLs (separated by comma)")
+	flag.StringVar(&URL, "s", DefaultNatsURL, "The nats server URLs (separated by comma)")
+	flag.StringVar(&URL, "server", DefaultNatsURL, "The nats server URLs (separated by comma)")
 	flag.StringVar(&clusterID, "c", "test-cluster", "The NATS Streaming cluster ID")
 	flag.StringVar(&clusterID, "cluster", "test-cluster", "The NATS Streaming cluster ID")
 	flag.StringVar(&clientID, "id", "", "The NATS Streaming client ID to connect with")
@@ -90,12 +90,11 @@ func main() {
 		usage()
 	}
 
-
-	sc, err := Connect(clusterID, clientID)
+	sc, err := Connect(clusterID, clientID, NatsURL(URL))
 	if err != nil {
-		log.Fatalf("Can't connect: %v.\nMake sure a NATS Streaming Server is running at: %s", err, opts.NatsURL)
+		log.Fatalf("Can't connect: %v.\nMake sure a NATS Streaming Server is running at: %s", err, URL)
 	}
-	log.Printf("Connected to %s clusterID: [%s] clientID: [%s]\n", opts.NatsURL, clusterID, clientID)
+	log.Printf("Connected to %s clusterID: [%s] clientID: [%s]\n", URL, clusterID, clientID)
 
 	subj, i := args[0], 0
 

--- a/stan.go
+++ b/stan.go
@@ -97,6 +97,14 @@ var DefaultOptions = Options{
 // Option is a function on the options for a connection.
 type Option func(*Options) error
 
+// NatsURL is an Option to set the URL the client should connect to.
+func NatsURL(u string) Option {
+	return func(o *Options) error {
+		o.NatsURL = u
+		return nil
+	}
+}
+
 // ConnectWait is an Option to set the timeout for establishing a connection.
 func ConnectWait(t time.Duration) Option {
 	return func(o *Options) error {

--- a/stan_test.go
+++ b/stan_test.go
@@ -1928,3 +1928,14 @@ func TestMaxPubAcksInflight(t *testing.T) {
 		t.Fatal("Should have blocked after 1 message sent")
 	}
 }
+
+func TestNatsURLOption(t *testing.T) {
+	s := RunServer(clusterName)
+	defer s.Shutdown()
+
+	sc, err := Connect(clusterName, clientName, NatsURL("nats://localhost:5555"))
+	if err == nil {
+		sc.Close()
+		t.Fatal("Expected connect to fail")
+	}
+}

